### PR TITLE
Add doesNotHaveToString assertion

### LINKED ISSUE #2028: Add doesNotHaveSameHashCodeAs and doesNotHaveToString assertions
#### Summary
I sugest to add _doesNotHaveSameHashCodeAs_ and _doesNotHaveToString_ assertions to reflect _isEqualTo_/_isNotEqualTo_. (With a shorter name maybe?)

#### Example

```java
assertThat(foo).isEqualTo(foo);
assertThat(foo).hasSameHashCodeAs(foo);
assertThat(foo).hasToString(foo.toString());

// Would replace:
assertThat(foo).isNotEqualTo(bar);
assertThat(foo.hashCode()).isNotEqualTo(bar.hashCode());
assertThat(foo.toString()).isNotEqualTo(bar.toString());

// By:
assertThat(foo).isNotEqualTo(bar);
assertThat(foo).doesNotHaveSameHashCodeAs(bar);
assertThat(foo).doesNotHaveToString(bar.toString());
```

### PR DESCRIPTION
**Pull request by lykims**
#### Check List:
* Fixes #2028 
* Unit tests : YES
* Javadoc with a code example (on API only) : YES

Minor fixes for `hasToString` doc description and `ShouldHaveToString` error as well.

### CODE REVIEW

File: src/main/java/org/assertj/core/api/Assert.java [hunk 1]
```diff
@@ -467,11 +467,30 @@
...
+   * Verifies that actual {@code actual.toString()} is not equal to the given {@code String}.
+   * <p>
+   * Example :
+   * <pre><code class='java'> CartoonCharacter homer = new CartoonCharacter("Homer");
+   *
+   * // Instead of writing ...
+   * assertThat(homer.toString()).isNotEqualTo("Marge");
+   * // ... you can simply write:
+   * assertThat(homer).doesNotHaveToString("Marge");
+   * </code></pre>
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/Assert.java:485]: join the two lines

File: src/main/java/org/assertj/core/error/ShouldNotHaveToString.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+package org.assertj.core.error;
+
+public class ShouldNotHaveToString extends BasicErrorMessageFactory {
+
+  public static ErrorMessageFactory shouldNotHaveToString(Object actual, String other) {
+    return new ShouldNotHaveToString(actual, other);
+  }
+
+  private ShouldNotHaveToString(Object actual, String other) {
+    super("%nExpecting actual's toString() not to be equal to:%n  <%s>%nbut was:%n  <%s>", other, actual.toString());
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/error/ShouldNotHaveToString.java:22]: let's simplify this to:
```
"%nExpecting actual's toString() not to be equal to:%n  <%s>"
```
the reason is `but was:%n  <%s>` section is actually repeating actual's `toString` value

File: src/test/java/org/assertj/core/internal/objects/Objects_assertDoesNotHaveToString_Test.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    objects.assertDoesNotHaveToString(someInfo(), actual, "bar");
+    objects.assertDoesNotHaveToString(someInfo(), actual, null);
+  }
+
+  @Test
+  void should_fail_if_actual_is_null() {
+    // GIVEN
+    Object actualObject = null;
+    // WHEN
+    ThrowingCallable code = () -> objects.assertDoesNotHaveToString(someInfo(), actualObject, "bar");
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/internal/objects/Objects_assertDoesNotHaveToString_Test.java:50]: use `expectAssertionError` from `org.assertj.core.util.AssertionsUtil` which returns an `AssertionError`

File: src/test/java/org/assertj/core/internal/objects/Objects_assertDoesNotHaveToString_Test.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+  @Test
+  void should_fail_if_actual_toString_is_equal_to_the_other_String() {
+    // GIVEN
+    AssertionInfo info = someInfo();
+    // WHEN
+    AssertionError error = expectAssertionError(() -> objects.assertDoesNotHaveToString(info, actual, "Person[name='foo']"));
+    // THEN
+    verify(failures).failure(info, shouldNotHaveToString("Person[name='foo']", "Person[name='foo']"));
+    assertThat(error).hasMessageContainingAll("Person[name='foo']");
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/internal/objects/Objects_assertDoesNotHaveToString_Test.java:63]: remove this line, it is already covered by `verify(failures)` (we don't want to test the error message but only that we call the proper error message factory)

File: src/test/java/org/assertj/core/internal/objects/Objects_assertDoesNotHaveToString_Test.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    // THEN
+    assertThatAssertionErrorIsThrownBy(code).withMessage(actualIsNull());
+  }
+
+  @Test
+  void should_fail_if_actual_toString_is_equal_to_the_other_String() {
+    // GIVEN
+    AssertionInfo info = someInfo();
+    // WHEN
+    AssertionError error = expectAssertionError(() -> objects.assertDoesNotHaveToString(info, actual, "Person[name='foo']"));
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/internal/objects/Objects_assertDoesNotHaveToString_Test.java:60]: remove `AssertionError error` local variable (see next comment why)

**Review by joel-costigliola — CHANGES_REQUESTED** [review]
@lykims thanks, good PR, just a few improvements to do and we are good.

**Comment by lykims** [comment]
@joel-costigliola I addressed your comments. I also applied changes suggested for `Objects_assertDoesNotHaveToString_Test` to `Objects_assertHasToString_Test`. Let me know if I miss anything!

**Review by joel-costigliola — APPROVED** [review]

**Comment by joel-costigliola** [comment]
Integrated thanks for the good work @lykims!

**Pull request merged**